### ISSUE TITLE
BCC list entry not expanded with empty "To:" address

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#167]
If we use an address book list name in the BCC field, this is not turned into the actual recipients on send, if we also leave the "to" field empty. AS a workaround, it is possible to generate a SmartTemplate that sets the BCC automatically like this:

`%header.set(bcc,"Friends <friends>")%`

Use the name of the list on the left and within the < angle brackets >, just like Tb would do it when it auto-fills the address field when you type the list name.

You can send off this email without filling the "to" field which achieves a list mail with undisclosed recipients. However as long as SmartTemplates (2.17) is active you cannot do this manually as Thunderbird will not expand the list as long as To is not filled and take this literally - which will lead to the mail server rejecting the email.

**Comment by RealRaven2000** [comment]
Here is a quick fix version. The problem was caused by "late deferred" variables that can be used when writing new mails and which update directly from the To:, Cc: and Subject: lines when hitting the "Send" button. The function `GetComposeDetails` is called to retrieve and update the fields, unforunately it also undoes any list expansions. I have now added a call to expand recipients (which Thundebrird does / did when hitting the send button:

```
        let composeDetails = GetComposeDetails(); // Refresh subject and address fields
        expandRecipients(); // [issue 167] - refresh lists!

```

Here is a test version:
[smartTemplate-fx-3.9.1pre10.zip](https://github.com/RealRaven2000/SmartTemplates/files/7959739/smartTemplate-fx-3.9.1pre10.zip)

to install, as always drag download the zip file above and drag directly into Thunderbird Add-ons manager (without unpacking)
